File: 20-BinarySearchProblems/1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int>mpp;
        vector<int>x;
        for(int i=0;i<nums.size(); i++) mpp[nums[i]]=i;
       for(int i = 0; i<nums.size(); i++) {
        int find = target - nums[i];
        if(mpp.find(find) != mpp.end() && mpp[find] != i) {
            return {i, mpp[find]};
        }
       }
        return {};
    }
};

File: 20-BinarySearchProblems/1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
       int[] arr = new int[2];
       LinkedHashMap<Integer,Integer>mpp = new LinkedHashMap<>();

       for(int i=0;i< nums.length;i++) {
            mpp.put(nums[i],i);
       }
    
       for(int i=0;i< nums.length;i++) {
            int t = target;

            if(mpp.containsKey(t - nums[i]) && mpp.get(t - nums[i]) != i) {
                arr[0] = i;
                arr[1] = mpp.get(t - nums[i]);
                return arr;
            }
       }
       return arr;
    }
}